slurm/slurm.py

```diff
@@ -1,16 +1,29 @@
+import glob
+import json
 import os
+import time
 import reframe as rfm
 import reframe.utility.sanity as sn
 
 
 os.environ["TEST_ENVAR_OUTSIDE"] = 'defined'
 
 affinity_script = """
+import json
 import os
-affinity = os.sched_getaffinity(0)
-print("num_cores: " + str(len(affinity)))
-for aff in affinity:
-    print("affinity: " + str(aff))"""
+import time
+affinity = list(os.sched_getaffinity(0))
+task_id = os.getenv("SLURM_PROCID", "0") + "_" + os.getenv("SLURM_STEP_ID", "0")
+json.dump(affinity, open(f"affinity{task_id}.json", "w"))
+time.sleep(20)
+"""
+
+
+def load_affinities():
+    affinities = []
+    for aff in glob.glob('affinity*.json'):
```

Comment:
and this happens in per test/per job directory so we're sure it's only the just created `affinity.....json` files?

Reply:
yes, each test runs in its own subdirectory under `stage` and this dir is automatically cleaned up before re-running the test.